getusers.php

```diff
@@ -194,17 +196,96 @@
 			$users = $DB->get_records_sql($sql, array($context_id, $course_id, $grouping_id));
 		}
 
+		/**
+		 * We already have course staff, so I just check to see if the current
+		 * user is a course_staff member, if so we add build the @all dataset.
+		 */
+		if(!empty($course_staff[$USER->id])) {
+			// Always get only class/group users
+			if(!empty(group_id)) {
+				$groupusersql = "
+				SELECT DISTINCT
+					ue.userid,
+					e.courseid,
+					u.firstname,
+					u.lastname,
+					u.username,
+					r.shortname
+				FROM {user_enrolments} ue
+				JOIN {enrol} e ON (e.id = ue.enrolid)
+				JOIN {user} u ON (ue.userid = u.id)
+				JOIN {role_assignments} ra ON (u.id = ra.userid AND ra.contextid = ?)
+				JOIN {role} r ON (ra.roleid = r.id)
+				JOIN {groups_members} gm ON (u.id = gm.userid)
+				JOIN {groupings_groups} gg ON (gm.groupid = gg.groupid)
+				WHERE e.courseid = ?
+				AND gm.groupid = ?
+				AND r.shortname = 'student'
+				ORDER BY firstname;";
+
+				$groupusers = $DB->get_records_sql($groupusersql, array($context_id, $course_id, $group_id));
+			}
+			
+			if(!empty($grouping_id)) {
+				$groupingusersql = "
+				SELECT DISTINCT
+					ue.userid,
+					e.courseid,
+					u.firstname,
+					u.lastname,
+					u.username,
+					r.shortname
+				FROM {user_enrolments} ue
+				JOIN {enrol} e ON (e.id = ue.enrolid)
+				JOIN {user} u ON (ue.userid = u.id)
+				JOIN {role_assignments} ra ON (u.id = ra.userid AND ra.contextid = ?)
+				JOIN {role} r ON (ra.roleid = r.id)
+				JOIN {groups_members} gm ON (u.id = gm.userid)
+				JOIN {groupings_groups} gg ON (gm.groupid = gg.groupid)
+				WHERE e.courseid = ?
+				AND gm.groupingid = ?
+				AND r.shortname = 'student'
+				ORDER BY firstname;";
+
+				$groupingusers = $DB->get_records_sql($groupingusersql, array($context_id, $course_id, $grouping_id));
+			}
+
+			if(!empty($groupusers) && !empty($groupingusers)) {
+				$grouped_users = array_merge($groupusers, $groupingusers);
+			} elseif(!empty($groupusers)) {
+				$grouped_users = $groupusers;
+			} elseif(!empty($groupingusers)) {
+				$grouped_users = $groupingusers;
+			}
+			
+		}
+
 		$users = array_merge($users, $course_staff);
+		$allUserIds = null;
```

Comment:
Can we initialize this to an empty version of the correct type? Like this looks like it's a string so initializing it to "".
`empty("")` is still `true`.

Reply:
cool :)